test/unit/Cardano/Wallet/CoinSelection/FeeSpec.hs

```diff
@@ -524,8 +557,8 @@ instance Arbitrary Address where
 instance {-# OVERLAPS #-} Arbitrary (Network -> Address) where
     shrink _ = []
     arbitrary = do
-        mainnetA <- genAddress (39, 43)
-        testnetA <- genAddress (46, 50)
+        mainnetA <- genAddress (33, 33)
+        testnetA <- genAddress (40, 40)
```

Comment:
This ought to be only the internal payload of the address. So, same confusion as within the fee calculation but, on the generation this time. The error was hard to catch because this one actually compensate for the previous one in the implementation. So, only in rare cases (1 tests out of 33k!), the double-counting would not be compensated by the wrong generation here and trigger an error!